project-base/src/Controller/Front/OrderController.php

```diff
@@ -213,7 +213,7 @@ public function indexAction()
         if ($isValid) {
             if ($orderFlow->nextStep()) {
                 $form = $orderFlow->createForm();
-            } elseif ($this->isFlashMessageBagEmpty()) {
+            } elseif (count($this->getErrorMessages()) === 0 && count($this->getInfoMessages()) === 0) {
```

Comment:
It seems really strange the order flow is dependent on FlashMessages.

Could not `$form->isValid()` or `count($form->getErrors) === 0` be used here instead?

---

Otherwise it still should be stopped only when errors appears.

```suggestion
            } elseif (count($this->getErrorMessages()) === 0) {
```

Reply:
You might want to display some success flash message and that would prevent Order from completion. Info messages might be used for situations like: product in your cart has new price, do you still want to finish the order?

Reply:
I sticked to original expectation:

> Expected result
>
> Order cannot be sent only with error flash message. Or order can be sent with info flash message.

I aggree you may do not want to continue in ordering when product price has changed. But should not even here be some kind of form validation instead of throwing flash message?

Reply:
This conversation feels unresolved. Going by guts feelings, I have to agree with @s3tezsky .... flash messages should not govern the logic. I do get the example you raised @TomasLudvik (price change) but message should not be the only indicator affecting the flow of order process. 
Lets get third prg opinion on this one please.

Reply:
I absolutely agree with @sslt and @s3tezsky . There should be some validation instead of checking flash messages.

Reply:
Order process must be postponed by anything with error/information for user. Validation in order form can not check everything in app. 

Or do you want create validator for empty flash message bag? How would you imagine such a validator?

Reply:
@henzigo this conversation remained unresolved even we updated issue and agreed with @s3tezsky and @pk16011990 that current solution solves the problem, we are closing this now